src/components/ui4/SwapDetailsCard.tsx

```diff
@@ -87,7 +87,8 @@ export function SwapDetailsCard(props: Props) {
   }
 
   // The wallet may have been deleted:
-  const destinationWallet = useSelector(state => state.core.account.currencyWallets[payoutWalletId])
+  const account = useSelector(state => state.core.account)
+  const destinationWallet = account.currencyWallets[payoutWalletId]
```

Comment:
No, selecting deeply would be a bug in this case! You are only allowed to select `state.core.account`, since that is what the reducer outputs. Selecting `state.core.account.anything_else` will not update correctly, since redux will not know if the account mutates its properties. A better approach is to do:

```ts
const account = useSelector(state => state.core.account)
const currencyWallets = useWatch(account, 'currencyWallets')
const destinationWallet = currencyWallets[payoutWalletId]
```

That way we subscribe to redux using `useSelector`, and we subscribe to the core using `useWatch`.

Reply:
Done in fixup